File: number-guessing-game-python/game/game.py
from utils.utils import checkGuess, getTarget, getUserInput, printMessage


class NumberGuessingGame:
    def __init__(self, number_range: int, mode: str):
        target = getTarget(10 ** (number_range - 1), (10**number_range) - 1)
        self.target = target
        health = {"E": 6, "M": 4, "H": 3}
        self.lives = health.get(mode.upper(), 6)

    def play(self):
        while self.lives > 0:
            printMessage(f"You have {self.lives} lives left")
            guess = getUserInput("Your Guess")
            if checkGuess(self.target, guess):
                printMessage("You Win!!!")
                break
            else:
                self.lives -= 1
                hint = "Guess lower" if guess > self.target else "Guess higher"
                printMessage(hint)
        if self.lives <= 0:
            printMessage("You have run out of lives")
            printMessage(f"The correct answer was {self.target}")

File: number-guessing-game-javascript/src/game.js
import { Utils } from "./utils.js";
export class NumberGuessingGame {
    constructor(number_range, lives) {
        this.number_range = number_range;
        this.lives = lives;
        this.ut = new Utils();
        this.target = this.ut.getTarget(number_range);
    }

    play() {
        while (this.lives > 0) {
            alert(`You have ${this.lives} lives left`);
            var guess = this.ut.getUserInput();
            if (this.ut.checkGuess(this.target, guess)) {
                alert("YOU WIN!!!");
                break;
            } else {
                this.lives -= 1;
                let hint;
                if (guess > this.target) {
                    hint = "Guess lower";
                } else {
                    hint = "Guess higher";
                }
                alert(`${hint}`);
            }
        }
        if (self.lives <= 0) {
            alert("You have run out of lives");
            alert(`The correct answer was ${this.target}`);
        }
    }
}
